### PR TITLE
UI/15399749 colony switcher

### PR DESCRIPTION
**Pull request by krawczykkatarzyna**
## Description

- [] TODO write description

**New stuff** ✨

* New stuff goes here

**Changes** 🏗

* Changes go here

**Deletions** ⚰️

* Deletions go here

## TODO

- [ ] Add TODOs

(Resolves | Contributes to) #31415

### CODE REVIEW

File: src/components/common/Extensions/ColonySwitcher/ColoniesDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import React, { FC } from 'react';
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/common/Extensions/ColonySwitcher/ColoniesDropdown.tsx:1]: Please, create a partials directory where you put smaller parts of this component

File: src/components/common/Extensions/ColonySwitcher/ColoniesDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import { Colony, WatchListItem } from '~types';
+import ColonyItem from './ColonyItem';
+import ColonyAvatar from '~shared/ColonyAvatar';
+import { IColoniesDropdown } from './types';
+import { useSelectedColony } from './hooks';
+
+const displayName = 'common.Extensions.ColonySwitcher.ColoniesDropdown';
+
+const ColoniesDropdown: FC<IColoniesDropdown> = ({ watchlist = [] }) => {
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/common/Extensions/ColonySwitcher/ColoniesDropdown.tsx:10]: Why do you use the `I` prefix in the interface name? We use `ComponentProps` convention in this project.

File: src/components/common/Extensions/ColonySwitcher/ColoniesDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    <div className="h-[24.75rem] p-1 w-full">
+      <div className="flex items-center p-2 md:ml-4">
+        <div className="flex mr-2">
+          <ColonyAvatar colony={colonyToDisplay as Colony} colonyAddress={colonyToDisplayAddress || ''} size="xxs" />
+        </div>
+        <div className="font-normal text-md text-gray-900">
+          {colonyToDisplay?.metadata?.displayName || colonyToDisplay?.name}
+        </div>
+      </div>
+      <div className="w-full h-[0.0625rem] bg-gray-200 my-2" />
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/common/Extensions/ColonySwitcher/ColoniesDropdown.tsx:31]: Shouldn't be a list?

File: src/components/common/Extensions/ColonySwitcher/ColonyAvatarWrapper.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        <Icon name="gnosis" />
+      </div>
+    </div>
+
+    {isMobile && (
+      <div className="font-semibold text-md text-gray-900">
+        {colonyToDisplay?.metadata?.displayName || colonyToDisplay?.name}
+      </div>
+    )}
+
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/common/Extensions/ColonySwitcher/ColonyAvatarWrapper.tsx:38]: Remove blank space

File: src/stories/common/ColonySwitcher.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+                </div>
+              )}
+              {isMobile && (
+                <ColonyDropdownMobile isOpen={isOpen} isMobile={isMobile} userLoading={userLoading}>
+                  {!!watchlistMock.length && <ColoniesDropdown watchlist={[...watchlistMock].sort(sortByDate)} />}
+                </ColonyDropdownMobile>
+              )}
+            </div>
+          )}
+
```
Comment (KatarzynaStefanowiczPagepro) [inline src/stories/common/ColonySwitcher.stories.tsx:104]: Remove blank space

File: src/components/common/Extensions/ColonySwitcher/ColonySwitcher.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+              {!!watchlist.length && !userLoading && <ColoniesDropdown watchlist={[...watchlist].sort(sortByDate)} />}
+            </ColonyDropdownMobile>
+          )}
+        </div>
+      )}
+
+      {isMobile && isOpen && (
+        <button
+          type="button"
+          aria-label="Close dropdown"
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/common/Extensions/ColonySwitcher/ColonySwitcher.tsx:104]: Use translation here

`aria-label={formatMessage({ id: 'nameHere' })}`

File: src/stories/common/ColonySwitcher.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+            offset: popperTooltipOffset,
+          },
+        },
+      ],
+    },
+  );
+
+  return (
+    <Router>
+      <div className="w-[26.75rem] md:64rem">
```
Comment (KatarzynaStefanowiczPagepro) [inline src/stories/common/ColonySwitcher.stories.tsx:58]: Why did you put the whole component here?
Is it not enough to return just a `<ColonySwitcher/>` here?

Comment (krawczykkatarzyna) [reply src/stories/common/ColonySwitcher.stories.tsx:58]: no it's won't be enough, API not works in storybook so, it's the only way to show list of colonies in dropdown

File: src/components/common/Extensions/ColonySwitcher/ColonySwitcher.tsx [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+          colonyToDisplayAddress={colonyToDisplayAddress}
+          setTriggerRef={setTriggerRef}
+        />
+      </button>
+
+      {/* @TODO: add wallet buttons */}
+
+      {isOpen && (
+        <div className="h-auto absolute top-[6.5rem] md:top-[2.3rem]">
+          {!isMobile && (
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/common/Extensions/ColonySwitcher/ColonySwitcher.tsx:76]:
```
{isMobile ? (
  <ColonyDropdownMobile isOpen={isOpen} isMobile={isMobile} userLoading={userLoading}>
    {!!watchlist.length && !userLoading && (
      <ColoniesDropdown watchlist={[...watchlist].sort(sortByDate)} isMobile={isMobile} />
    )}
  </ColonyDropdownMobile>
) : (
  <div
    ref={setTooltipRef}
    {...getTooltipProps({
      className: clsx('h-[24.75rem] p-1 flex justify-center z-[9999] tooltip-container', {
        'w-[26.75rem] border-none shadow-none': isMobile,
        'w-[15.1875rem]': !isMobile,
      }),
    })}
  >
    {userLoading && (
      <div className="h-[24.75rem] p-1 flex justify-center">
        <SpinnerLoader appearance={{ size: 'medium' }} />
      </div>
    )}
    {!!watchlist.length && !userLoading && <ColoniesDropdown watchlist={[...watchlist].sort(sortByDate)} />}
  </div>
)}
```

File: src/components/common/Extensions/ColonySwitcher/partials/ColoniesDropdown.tsx [hunk 9]
```diff
@@ -0,0 +1,58 @@
+import React, { FC } from 'react';
+import { Colony, WatchListItem } from '~types';
+import ColonyItem from './ColonyItem';
+import ColonyAvatar from '~shared/ColonyAvatar';
+import { ColoniesDropdownProps } from '../types';
+import { useSelectedColony } from '../hooks';
+
+const displayName = 'common.Extensions.ColonySwitcher.ColoniesDropdown';
```
Comment (KatarzynaStefanowiczPagepro) [inline src/components/common/Extensions/ColonySwitcher/partials/ColoniesDropdown.tsx:8]: 'common.Extensions.ColonySwitcher.partials.ColoniesDropdown';

**Review by KatarzynaStefanowiczPagepro — APPROVED** [review]

**Pull request merged**